### PR TITLE
Add ActiveModel::Serializer support

### LINKED ISSUE #70: Integrate with ActiveModel::Serializer
I think it's a fairly common use-case where you want Algolia search results to follow the same structure as the JSON responses from the application's API. One of the most popular ways to render JSON is to use [AMS](https://github.com/rails-api/active_model_serializers). Rather than having to define a serializer for an object as well as the `algoliasearch` block, it would be great if Algolia can utilize the object's serializer directly. This would also simplify dealing with models that have any `has_*` relationships.

### PR DESCRIPTION
**Pull request by redox**
Fix #70

### CODE REVIEW

**Comment by skunkworker** [comment]
+1 for this PR. This is very beneficial to maintain consistency between Algolia results and my API results.

**Comment by julienbourdeau** [comment]
I'm working on it, this is the next "big" thing I want to merge. Although, I'm currently running into a lot of build issue (for Ruby < 2.0) 😞

**Comment by julienbourdeau** [comment]
Yes, all the way down to 1.8.7 💪

File: lib/algoliasearch-rails.rb [hunk 1]
```diff
@@ -162,7 +166,11 @@ def attributes_to_hash(attributes, object)
 
     def get_attributes(object)
       attributes = if @attributes.nil? || @attributes.length == 0
-        get_default_attributes(object)
+        if @serializer.nil?
+          get_default_attributes(object)
+        else
+          {}
+        end
```
Comment (redox) [inline lib/algoliasearch-rails.rb]: I don't understand this change @julienbourdeau; also, it doesn't match https://github.com/algolia/algoliasearch-rails/pull/266/files#diff-dfaec67e907148ea8211e17340f0f30cL145

**Comment by ivancevich** [comment]
Hi guys, thanks for adding support to ActiveModel::Serializer!
Do you think this is going to be merged soon? I changed my Gemfile definition to use this branch for now, since I really need to use existing serializers.

**Comment by julienbourdeau** [comment]
Hi @ivancevich,

Thank you for testing the PR, this is very valuable 🙏 
Ideally I'd rather fix the build before merging but if I can't, I'll release it. I'm confident enough that it didn't break any existing feature.

Thanks a lot for your patience.

**Comment by Spone** [comment]
Where is this new feature documented? I think there should be something about it in the README :)

**Comment by julienbourdeau** [comment]
I haven't documented it yet, thank you for reminding me. I'll open a PR today on our documentation.
Note that the readme is generated from the docs.

**Comment by ivancevich** [comment]
Hi @julienbourdeau - I'm using the new `use_serializer` feature for a while now, and it's been great. Though, I've just found a problem with serializers that use the `has_one` association. For example:

```ruby
class SingleSerializer < ActiveModel::Serializer
  attributes :id,
    :artists,
    :genre,
    :slug

  has_one :release
  has_one :label
end
```

In the example above, all "attributes" work just fine. Though, both `:release` and `:label` are not getting indexed in Algolia. I'm not sure why.

**Comment by Spone** [comment]
> I haven't documented it yet, thank you for reminding me. I'll open a PR today on our documentation.
> Note that the readme is generated from the docs.

ping :)

**Pull request merged**